chef/cookbooks/crowbar/recipes/default.rb

```diff
@@ -257,6 +257,21 @@
     group node[:apache][:group]
     mode "0640"
   end
+
+  client_users = users.dup
+  node["crowbar"]["client-users"].each do |k, h|
+    next if h["disabled"]
+    # Fix passwords into digests.
+    h["digest"] = Digest::MD5.hexdigest("#{k}:#{realm}:#{h["password"]}") if h["digest"].nil?
+    client_users[k] = h
+  end
```

Comment:
Why do you need to create `client_users`? Can't you just pass same `users` argument to the template when creating htdigest-clients?

Reply:
The `users` variable doesn't contain the client user, and if it did that would mean the client machines could access `/`. So we need a separate variable that adds the client user as well as the other users to this new htdigest file.

Reply:
Ah, now I know why I was confused, it's that `client_users = users.dup`. Is there a reason for that? Would it be better to initialize it as empty hash?

Reply:
If it was initialized to an empty hash then we would still need to add the crowbar and machine-install user to it, as those users also need to use the transition API. Better to just copy the existing one.

Reply:
Thanks for explanation